### PR TITLE
Move web worker and types into @pmndrs/cannon-worker-api package

### PR DESCRIPTION
**Pull request by isaac-mason**
- Move setup types and body props types into `@pmndrs/cannon-worker-api` package
- Move cannon web worker into `@pmndrs/cannon-worker-api` package
- Update `@react-three/cannon` to use `@pmndrs/cannon-worker-api`

### CODE REVIEW

File: packages/cannon-worker-api/src/setup.ts [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+}
+export type RayhitEvent = Omit<WorkerRayhitEvent['data'], 'body'> & { body: Object3D | null }
+
+type CannonEvent = CollideBeginEvent | CollideEndEvent | CollideEvent | RayhitEvent
+type CallbackByType<T extends { type: string }> = {
+  [K in T['type']]?: T extends { type: K } ? (e: T) => void : never
+}
+
+// todo - check if used
+export type CannonEvents = { [uuid: string]: Partial<CallbackByType<CannonEvent>> }
```
Comment (bjornstar) [inline packages/cannon-worker-api/src/setup.ts:107]: this is used in ProviderContext, can move it there.

Comment (isaac-mason) [reply packages/cannon-worker-api/src/setup.ts:107]: Ah thanks, will move

**Comment by bjornstar** [comment]
> ### TODO
> * Change `@pmndrs/cannon-worker-api` body types to use `shape` instead of `args`
>   
>   * Have `@react-three/cannon` shapes use `args` for backwards compatibility
>   * Avoiding using `args` in `@pmndrs/cannon-worker-api` as it is a r3f pattern
> * Only re-export types from `@pmndrs/cannon-worker-api` in `@react-three/cannon`

Let's start with getting `@pmndrs/cannon-worker-api` released before we start iterating on the API.

**Comment by isaac-mason** [comment]
Okay sounds good. I'll just move a couple more types around then mark the PR as ready.

File: packages/cannon-worker-api/src/setup.ts (hunk 1 above)
Comment (isaac-mason) [reply packages/cannon-worker-api/src/setup.ts]: Moved to `@react-three/cannon` `Provider.tsx` 👍

**Comment by bjornstar** [comment]
Cool, let me publish the package.

**Pull request merged**